Windows UI tests (most of them)

ericmehl opened the pull request:
This gets the Gaffer*UI tests passing, with the exception of the renderer tests (Arnold, Cycles), which need some more work.

### Breaking changes ###

None

### Checklist ###

- [X] I have read the [contribution guidelines](https://github.com/GafferHQ/gaffer/blob/main/CONTRIBUTING.md).
- [X] I have updated the documentation, if applicable.
- [X] I have tested my change(s) in the test suite, and added new test cases where necessary.
- [X] My code follows the Gaffer project's prevailing coding style and conventions.

johnhaddon commented on python/GafferUI/ReferenceUI.py:
```diff
@@ -157,14 +157,14 @@ def _waitForFileName( initialFileName="", parentWindow=None ) :
+		return None
 
-	return str( path )
+	return path.path()
```
```AttributeError: 'FileSystemPath' object has no attribute 'path'```

I think this needs to be `standardPath()` after the tweak we made on the other PR.

ericmehl replied:
Ack! Blindly copying my old commit faux pas. Fixed in https://github.com/GafferHQ/gaffer/pull/5071/commits/1f138bf1c6f7814e7161244f39792cc622dc08e9

johnhaddon commented on python/GafferUITest/GraphGadgetTest.py:
```diff
@@ -1813,6 +1813,8 @@ def testDirtyTrackingForInitialFocusNode( self ) :
 		# Initially we expect the left branch of the switch to be highlighted.
 
 		window.setVisible( True )
+		self.waitForIdle( 1000 )
```
I'm surprised that this is necessary, because `assertHighlighting()` goes to reasonable lengths to wait for the highlighting to take effect.

ericmehl replied:
I could move the increased wait count to https://github.com/GafferHQ/gaffer/blob/0c868cc20f7c72930d6ce6048342ce3a7ac9a07d/python/GafferUITest/GraphGadgetTest.py#L1784 as an alternative. I guess that the default wait count of 1 is not enough for grandpa Windows to set things in motion?

ericmehl replied:
Offline we decided to leave this in place since it's not too much of an extra delay, and the proper solution is pretty involved for a one line fix.

johnhaddon commented on python/GafferUITest/GLWidgetTest.py:
```diff
@@ -75,7 +75,7 @@ def testOverlayWidgetAt( self ) :
 		w.setVisible( True )
 
-		self.waitForIdle( 10000 )
+		self.waitForIdle( 1000000 )
```
This test now takes a full 8 seconds on my Linux machine. Do we have any idea why we need to wait that long?

ericmehl replied:
Dropped in the latest push.

johnhaddon commented on python/GafferUITest/WidgetTest.py:
```diff
@@ -415,7 +415,7 @@ def testWidgetAt( self ) :
 		w2.setPosition( imath.V2i( 300 ) )
 
-		self.waitForIdle( 10000 )
+		self.waitForIdle( 1000000 )
```
As above, this is now taking the best part of 8 seconds. I'd really like to keep the test suite as fast to run as possible...

johnhaddon replied:
Oh, I see these tests are the same ones that are disabled in a later commit. Could we just drop this commit then, to keep things quick on Linux?

ericmehl replied:
Also dropped in the latest push.

johnhaddon reviewed (commented):
Nice one Eric!

Looks like we've got a CI failure on Windows still though :

```
"Error(s) running Gaffer"
testGadgetsAt (GafferUITest.ViewportGadgetTest.ViewportGadgetTest) ... 
Error: Process completed with exit code 1.
```

Am I right in thinking that this output indicates a hard crash? Since `gadgetsAt()` is clearly working OK outside of CI (non-mesa) I think it would be reasonable to disable that test temporarily on Windows in order to get the benefits of running Windows CI on everything else.

ericmehl commented:
My latest push addressed the comments above. It also skips an additional test failure in https://github.com/GafferHQ/gaffer/pull/5071/commits/356e96d66bab3e4f67333b517906ae7fb3f33eea and fixes a holdover from #5049 in https://github.com/GafferHQ/gaffer/pull/5071/commits/88dcfda922cc6e42069d9949677326995108a56c.

Ready for a new review.
